# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskSchedulerMainThread.cpp
# Commit: 79b3b0148eb4
# Full Hash: 79b3b0148eb416e0690d3e71723bd68ac67245e7
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Implement scheduler.yield r=jjaschke,dom-core,webidl,smaug
#   
#   Spec: https://wicg.github.io/scheduling-apis/#dom-scheduler-yield
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D234232
# ==============================================================================

diff -r 78181c32bdc6 -r 79b3b0148eb4 dom/webscheduling/WebTaskSchedulerMainThread.cpp
--- a/dom/webscheduling/WebTaskSchedulerMainThread.cpp	Thu Feb 27 16:17:35 2025 +0000
+++ b/dom/webscheduling/WebTaskSchedulerMainThread.cpp	Thu Feb 27 16:17:35 2025 +0000
@@ -5,7 +5,6 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#include "mozilla/TaskController.h"
 #include "mozilla/dom/TimeoutManager.h"
 
 #include "nsContentUtils.h"
@@ -23,8 +22,8 @@
   return NS_OK;
 }
 
-nsresult WebTaskSchedulerMainThread::SetTimeoutForDelayedTask(WebTask* aTask,
-                                                              uint64_t aDelay) {
+nsresult WebTaskSchedulerMainThread::SetTimeoutForDelayedTask(
+    WebTask* aTask, uint64_t aDelay, EventQueuePriority aPriority) {
   JSContext* cx = nsContentUtils::GetCurrentJSContext();
   if (!cx) {
     return NS_ERROR_UNEXPECTED;
@@ -33,7 +32,7 @@
   MOZ_ASSERT(global);
 
   RefPtr<DelayedWebTaskHandler> handler =
-      new DelayedWebTaskHandler(cx, this, aTask);
+      new DelayedWebTaskHandler(cx, this, aTask, aPriority);
 
   int32_t delay = aDelay > INT32_MAX ? INT32_MAX : (int32_t)aDelay;
 
@@ -44,12 +43,13 @@
                    Timeout::Reason::eDelayedWebTaskTimeout, &handle);
 }
 
-bool WebTaskSchedulerMainThread::DispatchEventLoopRunnable() {
+bool WebTaskSchedulerMainThread::DispatchEventLoopRunnable(
+    EventQueuePriority aPriority) {
   RefPtr<WebTaskMainThreadRunnable> runnable =
       new WebTaskMainThreadRunnable(this);
 
-  MOZ_ALWAYS_SUCCEEDS(NS_DispatchToMainThreadQueue(runnable.forget(),
-                                                   EventQueuePriority::Normal));
+  MOZ_ALWAYS_SUCCEEDS(
+      NS_DispatchToMainThreadQueue(runnable.forget(), aPriority));
   return true;
 }
 }  // namespace mozilla::dom